pkg/exchange/coinbase/exchange.go

```diff
@@ -201,21 +201,27 @@ func (e *Exchange) SubmitOrder(ctx context.Context, order types.SubmitOrder) (cr
 		req.ClientOrderID(order.ClientOrderID)
 	}
 
-	timeNow := time.Now()
+	// Record metrics timing
+	requestTime := time.Now()
 	res, err := req.Do(ctx)
+	duration := time.Since(requestTime)
+	recordOrderSubmissionMetrics(order, duration, err)
```

Comment:
why do you need to pass error

Reply:
Also, I need the error to get the response status code.
So that I can update corresponding metrics.